src/ripple/app/tx/impl/CreateOffer.cpp

```diff
@@ -722,6 +723,58 @@ CreateOffer::applyGuts (ApplyView& view, ApplyView& view_cancel)
 
     if (result == tesSUCCESS)
     {
+        // If a tick size applies, round the offer to the tick size
+        auto const& uPaysIssuerID = saTakerPays.getIssuer ();
+        auto const& uGetsIssuerID = saTakerGets.getIssuer ();
+
+        std::uint8_t uTickSize = Quality::maxTickSize;
+        if (!isXRP (uPaysIssuerID))
+        {
+            auto const sle =
+                view.read(keylet::account(uPaysIssuerID));
+            if (sle && sle->isFieldPresent (sfTickSize))
+                uTickSize = std::min (uTickSize,
+                    (*sle)[sfTickSize]);
+        }
+        if (!isXRP (uGetsIssuerID))
+        {
+            auto const sle =
+                view.read(keylet::account(uGetsIssuerID));
+            if (sle && sle->isFieldPresent (sfTickSize))
+                uTickSize = std::min (uTickSize,
+                    (*sle)[sfTickSize]);
+        }
+        if (uTickSize < Quality::maxTickSize)
+        {
+            auto const rate =
+                Quality{saTakerGets, saTakerPays}.round
+                    (uTickSize).rate();
+
+            // We round the side that's not exact,
+            // just as if the offer happened to execute
+            // at a slightly better (for the placer) rate
+            if (bSell)
+            {
+                // this is a sell, round the amount we pay
```

Comment:
From the perspective of the placer, isn't this rounding the amount we get? That is `takerPays` = `placerGets`.